### PR TITLE
Return proper responses for SCIM provisioned identities

### PR DESCRIPTION
**Pull request by tenyo**
This PR adds return responses to `ListSCIMProvisionedIdentities` and `GetSCIMProvisioningInfoForUser`, as before it was not possible to get the output of those calls.
Fixes https://github.com/google/go-github/issues/2338

### CODE REVIEW

File: github/scim.go [hunk 1]
```diff
@@ -62,17 +82,22 @@ type ListSCIMProvisionedIdentitiesOptions struct {
...
-		return nil, err
+		return nil, nil, err
 	}
-	return s.client.Do(ctx, req, nil)
+	identities := new(SCIMProvisionedIdentities)
+	resp, err := s.client.Do(ctx, req, identities)
+	if err != nil {
+		return nil, resp, err
+	}
+	return identities, resp, nil
```
Comment (gmlewis) [inline github/scim.go:100]: Let's please clean this up a bit while we're here and add a blank line before lines 100, 95, and 91.

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @tenyo !
Just a couple minor nits, please, and then we should be ready for a second LGTM+Approval from any other contributor to this repo.

File: github/scim.go (hunk 1 above)
Comment (tenyo) [reply github/scim.go:100]: Sure, I can do that - I actually removed the blank lines on purpose to match the formatting in the other functions.

Comment (tenyo) [reply github/scim.go:103]: OK, I've fixed those, let me know if you want me to add similar blank lines in the rest of the funcs in this file, so they all match.

Comment (gmlewis) [reply github/scim.go:103]: > OK, I've fixed those, let me know if you want me to add similar blank lines in the rest of the funcs in this file, so they all match.

You can if you would like to... Thank you!

Comment (tenyo) [reply github/scim.go:103]: All done, much better now ☺️  
You can probably add a linter rule in `.golangci.yml` to enforce that but there might be a bunch of files to fix 😅

**Review by wesleimp — APPROVED** [review]

**Comment by gmlewis** [comment]
Thank you, @wesleimp !
Merging.

**Pull request merged**